src/core/public/chrome/ui/header/collapsible_nav_group_enabled.tsx

```diff
@@ -0,0 +1,347 @@
+    e: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+    group: NavGroupItemInMap
+  ) => {
+    const fulfilledLinks = fulfillRegistrationLinksToChromeNavLinks(
+      navGroupsMap[group.id]?.navLinks,
+      navLinks
+    );
+    setCurrentNavGroup(group.id);
+
+    // the `navGroupsMap[group.id]?.navLinks` has already been sorted
+    const firstLink = fulfilledLinks[0];
+    if (firstLink) {
+      const propsForEui = createEuiListItem({
+        link: firstLink,
+        appId,
+        dataTestSubj: 'collapsibleNavAppLink',
+        navigateToApp,
+      });
+      propsForEui.onClick(e);
+    }
+  };
+
+  const navLinksForRender: ChromeNavLink[] = useMemo(() => {
+    if (currentNavGroup) {
+      return fulfillRegistrationLinksToChromeNavLinks(
+        navGroupsMap[currentNavGroup.id].navLinks || [],
+        navLinks
+      );
+    }
+
+    const visibleUseCases = Object.values(navGroupsMap).filter(
+      (group) => group.type === undefined && group.status !== NavGroupStatus.Hidden
+    );
+
+    if (visibleUseCases.length === 1) {
+      return fulfillRegistrationLinksToChromeNavLinks(
+        navGroupsMap[visibleUseCases[0].id].navLinks || [],
+        navLinks
+      );
+    }
+
+    const navLinksForAll: ChromeRegistrationNavLink[] = [];
+
+    // Append all the links that do not have use case info to keep backward compatible
+    const linkIdsWithUseGroupInfo = Object.values(navGroupsMap).reduce((total, navGroup) => {
+      return [...total, ...navGroup.navLinks.map((navLink) => navLink.id)];
+    }, [] as string[]);
+    navLinks
+      .filter((link) => !linkIdsWithUseGroupInfo.includes(link.id))
+      .forEach((navLink) => {
+        navLinksForAll.push(navLink);
+      });
+
+    // Append all the links registered to all use case
+    navGroupsMap[ALL_USE_CASE_ID]?.navLinks.forEach((navLink) => {
+      navLinksForAll.push(navLink);
+    });
+
+    // Append use case section into left navigation
+    Object.values(navGroupsMap)
+      .filter((group) => !group.type)
+      .forEach((group) => {
+        const categoryInfo = {
+          id: group.id,
+          label: group.title,
+          order: group.order,
+        };
+        const linksForAllUseCaseWithinNavGroup = group.navLinks
+          .filter((navLink) => navLink.showInAllNavGroup)
+          .map((navLink) => ({
+            ...navLink,
+            category: categoryInfo,
+          }));
+
+        navLinksForAll.push(...linksForAllUseCaseWithinNavGroup);
+
+        if (linksForAllUseCaseWithinNavGroup.length) {
+          navLinksForAll.push({
+            id: group.navLinks[0].id,
+            title: titleForSeeAll,
+            order: Number.MAX_SAFE_INTEGER,
+            category: categoryInfo,
+          });
+        }
+      });
+
+    return fulfillRegistrationLinksToChromeNavLinks(navLinksForAll, navLinks);
+  }, [navLinks, navGroupsMap, currentNavGroup]);
+
+  const width = useMemo(() => {
+    if (!isNavOpen) {
+      return 50;
```

Comment:
instead of using magic number, can we give it a name, is it percentage or other unit?

Reply:
Sure, will optimize this part.